### PR TITLE
refactor(ngx-components)!: completely switch to standalone

### PR DESCRIPTION
**Pull request by d-koppenhagen**
## Proposed changes

**BREAKING CHANGE:**
All Angular components are `standalone` now.
SCAM (Single Component Angular Modules) have been removed.
please import the standalone components directly in your `imports` section.

```ts app.component.ts
//app.component.ts
import { DBButton } from '@db-ui/ngx-components';

@Component({
  // ...
  imports: [
    // ...,
    DBButton
  ],
  // ...
})
```

closes [#237](https://github.com/db-ui/mono/issues/237)

## Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [x] Refactoring (fix on existing components or architectural decisions)
-   [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [x] Documentation Update (if none of the other choices apply)

## Further comments

This PR includes https://github.com/db-ui/mono/pull/2183 as well and sits on top of it.

### CODE REVIEW

File: packages/components/mitosis.config.js [hunk 1]
```diff
@@ -7,7 +7,8 @@ module.exports = {
 			typescript: true
 		},
 		angular: {
-			typescript: true
+			typescript: true,
+			standalone: true
```
Comment (nmerget) [inline packages/components/mitosis.config.js:11]: There is a `mitosis-angular.config.js` besides this file, which need the `standalone:true` as well

**Comment by d-koppenhagen** [comment]
Not sure why the playwright tests are failing. The want to access the route via HasLocation (e. g. `/#/04/accordion`). This locations strategy works when running the app in dev mode and when building+serving it locally with production build options: `npm run build && npm run preview`.
The strange thing is: when starting playwright locally in UI mode (`--ui`), it seems like an old version of the angular app is started and it uses the "normal" path location strategy. In fact, it always checks against the root route since the child routes are not loaded. Any idea?

File: showcases/angular-showcase/src/app/components/navigation-item/navigation-item.component.html [hunk 2]
```diff
@@ -19,31 +19,33 @@
...
 		>
 			<ng-container *dbNavigationContent>
-				<ng-container *ngIf="exampleProps.areaPopup">{{
-					exampleName
-				}}</ng-container>
-				<a href="#" *ngIf="!exampleProps.areaPopup">{{
-					exampleName
-				}}</a>
+				@if (exampleProps.areaPopup) {
+					{{ exampleName }}
```
Comment (nmerget) [inline showcases/angular-showcase/src/app/components/navigation-item/navigation-item.component.html:23]: Based on the snapshots, the exampleName is missing:

neutral-actual

Comment (d-koppenhagen) [reply showcases/angular-showcase/src/app/components/navigation-item/navigation-item.component.html:23]: Ah, I see. Now I also managed to run it locally (I forgot to re-build the outputs). I am on it

**Comment by nmerget** [comment]
> Not sure why the playwright tests are failing. The want to access the route via HasLocation (e. g. `/#/04/accordion`). This locations strategy works when running the app in dev mode and when building+serving it locally with production build options: `npm run build && npm run preview`. The strange thing is: when starting playwright locally in UI mode (`--ui`), it seems like an old version of the angular app is started and it uses the "normal" path location strategy. In fact, it always checks against the root route since the child routes are not loaded. Any idea?

I guess the tests are working only some of them are failing: https://github.com/db-ui/mono/actions/runs/7826104309/job/21351476929?pr=2184#step:10:20

It's only the navigation-item that has an issue. If the routing doesn't work in the CI all Tests would fail.

File: packages/components/mitosis.config.js (hunk 1 above)
Comment (d-koppenhagen) [reply packages/components/mitosis.config.js:11]: I though this too, but it works when only setting it in this file? What's the other file for?

**Review by nmerget — APPROVED** [review]

**Pull request merged**